vertx-web/src/main/java/io/vertx/ext/web/templ/impl/CachingTemplateEngine.java

```diff
@@ -42,6 +42,9 @@ protected String adjustLocation(String location) {
     if (!location.endsWith(extension)) {
       location += extension;
     }
+    if (location.endsWith("/" + extension)) {
```

Comment:
I believe this check should happen before, and only assert for:

```java
if (location.endsWith("/")) { ...
```
This way we can differentiate from request like this:
```
/.ext -> remains untouched
/ -> gets upgraded to /index.ext
/file -> gets upgraded to /file.ext
```